Add the size primary to `wash find`

ekinanp opened the pull request:
The size primary was modeled off its counterpart in BSD's `find`
command.

Signed-off-by: Enis Inan <enis.inan@puppet.com>

ekinanp commented on cmd/find/sizePrimary.go:
```diff
@@ -0,0 +1,112 @@
+			// required by the user's specified primaries when those primaries are or'ed
+			// with other primaries that could potentially return true for that entry. Do
+			// we ignore them? Do we still apply the predicate on them, even if that predicate
+			// may return true? We should figure this out soon.
```
I'm thinking it'd be a better UX if `wash find`'s predicate always returned false for entries that do not have the attribute(s) specified by the given expression (so something like `wash find /aws -size +1 -o true` will never display EC2 instances because size doesn't make sense for them). 

We don't have to enforce this for _all_ attributes, e.g. I think entries without `ctime`, `mtime` or `atime` attributes should still be processed. Hmm, maybe we shouldn't enforce it for `ctime`, but we should for the other attributes?

If the TODO here is not an issue or is something we should address if users become annoyed by it, I'm happy to remove it too.

MikaelSmith replied:
Why would you use `-o true`? I'm not clear on that.

ekinanp replied:
Oh, the real use-case could be something llike `find /aws -size +1 -o mtime +1h` where an entry without a size attribute but w/ an mtime attribute could satisfy the predicate

MikaelSmith replied:
I'm comfortable saying that predicate returns false, but not going any further. We can do something more complicated if specific examples come up.

ekinanp replied:
Sounds good, I'll remove the TODO.

MikaelSmith commented on cmd/find/sizePrimary.go:
```diff
@@ -0,0 +1,112 @@
+// where c => character (byte), k => kilobyte, M => megabyte, G => gigabyte, T => terabyte, P => petabyte
+//
+// Example inputs:
+//   -size 2   (true if the entry's size in 512-byte blocks, rounded up, is 2)
```
Also, why 512-byte blocks? 4096 is a more common default for Linux.

MikaelSmith replied:
I think you missed the table on the right from my link. mebibyte, gibibyte, tebibyte, pebibyte.

MikaelSmith replied:
Also line 39 isn't correct.

ekinanp replied:
Oh, you wanted all of them to be labeled as such. Sure, will update.

MikaelSmith replied:
I want to follow the ISQ standard. I'm ok with the single-character form referring to the colloquial kilobyte (now kibibyte), etc. We could later differentiate with `KB` and `KiB`, which are the official IEC symbols.

MikaelSmith commented on cmd/find/sizePrimary.go:
```diff
@@ -0,0 +1,112 @@
+
+	cmp := v[0]
+	if cmp == '+' || cmp == '-' {
+		v = strings.TrimPrefix(v, string(cmp))
```
So this allows `+++1`? Any reason to do that rather than just a string slice `v[1:]`?

ekinanp replied:
It doesn't -- `sizeValueRegex` already validated v. That's a good test-case to add though.

Ah yeah, string slice is simpler. Will update.

MikaelSmith replied:
Did string slicing work?

ekinanp commented:
@MikaelSmith Updated the PR (and yes, string slicing worked). Only unaddressed feedback is whether we should use 512/4096 byte blocks as the default. Note that the Linux `find` command also defaults to 512 byte blocks.

MikaelSmith commented:
Leaving it 512 is fine.
